csrc/multidevice/utils.h

```diff
@@ -123,9 +123,10 @@ int64_t requestedNumberOfDevices(Fusion*);
 void unshard(Fusion*);
 void unshard(TensorView*);
 
-// Returns the index of the a sharded axis if none return -1.
-// TODO: Assumes no merges/splits on sharded axis.
-int64_t getShardedAxis(TensorView*);
+// Returns the index of the sharded logical axis that produces the allocation
+// IterDomain sharded on `parallel_type`. If `tv` isn't sharded on the parallel
+// type, returns -1.
+int64_t getShardedAxis(const TensorView* tv, ParallelType parallel_type);
```

Comment:
nit:
```suggestion
int64_t getShardedLogicalAxis(const TensorView* tv, ParallelType parallel_type);
```

Reply:
Can you give some more details about what this function is supposed to return? As mentioned in the implementation, there's no single logical ID if the parallelized ID is a merge output. I don't have a good overall picture, so it's unclear to me what the context is.

Reply:
+1
I am also unsure about how this function is supposed to be used.

Reply:
Done!